pages/api/_middleware.ts

```diff
@@ -1,23 +1,45 @@
-import type { NextApiRequest } from 'next'
-import { getToken } from 'next-auth/jwt'
-import { NextResponse } from 'next/server'
+import { IncomingHttpHeaders } from "http2";
+import type { NextApiRequest } from "next";
+import { getToken } from "next-auth/jwt";
+import { NextResponse } from "next/server";
+
+interface CustomHeader {
+  get: (key: string) => string
+}
+
+type CombinedHeader = IncomingHttpHeaders & CustomHeader;
 
 const testnet = process.env.NEXT_E2E_TESTNET === "true";
 
 const blackList = {
-  GET: ['auth', 'past-events', 'seo', 'users', 'graphql'],
-  POST: ['issue']
-}
-
+  POST: ["network", "pull-request", "repos"],
+  PUT: ["issue/working", "network", "pull-request/review"],
+  DELETE: ["pull-request", "user", "repos"],
+  PATCH: ["user/connect"]
+};
 
+const UnauthorizedResponse = 
+  (reason: string) => new Response(JSON.stringify({ reason }), { status: 401, statusText: "Unauthorized" });
 
 export async function middleware(req: NextApiRequest) {
   const method = req.method;
-  const shouldCheckForToken = blackList[method]?.some(path => req.url.replace('/api/', '').search(path) > -1);
+  const endpoint = req.url.split("api/")[1];
+
+  const shouldCheckForToken = blackList[method]?.some(path => (new RegExp(`^${path}$`)).test(endpoint) );
```

Comment:
be less restrictive on the regex, and you can forgo the assignment to `endpoint = ...`, remove the `^` from this regex and we stop having to worry about the "/api/" part